src/badge.jsx

```diff
@@ -1,9 +1,54 @@
 import React from 'react';
 import Typography from './styles/typography';
 import getMuiTheme from './styles/getMuiTheme';
-import StylePropable from './mixins/style-propable';
 
-// Badge
+function getStyles(props, state) {
+  const {
+    primary,
+    secondary,
+  } = props;
+
+  const {
+    badge,
+  } = state.muiTheme;
+
+  const [
```

Comment:
Haha! :sweat_smile: :sweat_smile:  That monster (I agree it's a monster) was an attempt to make the previous ternary simpler:

```js
const badgeBackgroundColor = this.props.primary
  ? theme.primaryColor
  : this.props.secondary
    ? theme.secondaryColor
    : theme.color;

const badgeTextColor = this.props.primary
  ? theme.primaryTextColor
  : this.props.secondary
    ? theme.secondaryTextColor
    : theme.textColor;
```

I can switch it to use `let` + `if/else` even though `lets` make me sad :grin:  Though I agree, ternarys beyond one level deep are too complicated.